index.html

```diff
@@ -588,80 +635,106 @@ <h3>
+                global task=] on the [=networking task source=] using |global| to [=reject=]
+                |promise| with an {{"InvalidAccessError"}} {{DOMException}} and terminate these
+                steps.
+                </li>
+              </ol>
             </li>
-          </ol>
-        </li>
-        <li>Let |registration:ServiceWorkerRegistration| be [=this=]'s associated <a>service worker
-        registration</a>.
-        </li>
-        <li>If |registration|'s [=service worker registration/active worker=] is null, [=queue a
-        global task=] on the [=networking task source=] using |global| to [=reject=] |promise| with
-        an {{"InvalidStateError"}} {{DOMException}} and terminate these steps.
-        </li>
-        <li>Let |permission| be [=request permission to use=] "push".
-        </li>
-        <li>If |permission| is {{PermissionState/"denied"}}, [=queue a global task=] on the [=user
-        interaction task source=] using |global| to [=reject=] |promise| with a
-        {{"NotAllowedError"}} {{DOMException}} and terminate these steps.
-        </li>
-        <li>If |registration| has a <a>push subscription</a>:
-          <ol>
-            <li>Let |subscription| be the result of obtaining |registration|'s <a>push
-            subscription</a>. If there is an error, [=queue a global task=] on the [=networking
-            task source=] using |global| to [=reject=] |promise| with an {{"AbortError"}}
-            {{DOMException}} and terminate these steps.
+            <li>Let |subscription| be null.
+            </li>
+            <li>If |scope| is non-null:
+              <ol>
+                <li>If there is a [=push subscription=] with a [=push
+                subscription/window-accessible scope=] whose [=push subscription/scope=]
+                [=URL/equals=] |scope|, then set |subscription| to that [=push subscription=].
+                </li>
+              </ol>
             </li>
-            <li>Compare the |options| argument with the `options` attribute of |subscription|. The
-            contents of {{BufferSource}} values are compared for equality rather than
-            [=ECMAScript/reference record|reference=].
+            <li>Otherwise:
+              <ol>
+                <li>[=/Assert=]: |registration| is non-null.
+                </li>
+                <li>If |registration|'s [=service worker registration/active worker=] is null,
+                [=queue a global task=] on the [=networking task source=] using |global| to
+                [=reject=] |promise| with an {{"InvalidStateError"}} {{DOMException}} and terminate
+                these steps.
+                </li>
+                <li>If |registration|'s [=associated push subscription=] is non-null, then set
+                |subscription| to |registration|'s [=associated push subscription=].
+                </li>
+                <li>Set |scope| to |registration|'s [=service worker registration/scope URL=].
```

Comment:
If the scope URL is the root path then this is also accessible from window, right? I assume that's the intention.

Reply:
That's right.